apps/web/src/pages/quick-start/components/demo-flow/nodes/NodeStep.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useState } from 'react';
+import styled from '@emotion/styled';
+import { Popover } from '@mantine/core';
+
+import { useStyles } from '../../../../../design-system/template-button/TemplateButton.styles';
+import { colors, shadows, Text } from '../../../../../design-system';
+import { guidePreview, GuideTitleEnum, IBeat, IGuide } from './consts';
+import { useSegment } from '../../../../../components/providers/SegmentProvider';
+import { When } from '../../../../../components/utils/When';
+
+export function NodeStep({
+  data,
+  id,
+  Handlers,
+  Icon,
+  ActionItem,
+}: {
+  data: any;
+  id: string;
+  Handlers: React.FC<any>;
+  Icon: React.FC<any>;
+  ActionItem?: React.FC<any>;
+}) {
+  const { theme } = useStyles();
+  const { counter } = useCounter();
+  const [sequence, setSequence] = useState<IBeat>();
+
+  const label = data.label.toLowerCase();
+  const popoverData = guidePreview[label] as IGuide;
+  const titleGradient =
+    popoverData.title === GuideTitleEnum.DIGEST_PREVIEW || popoverData.title === GuideTitleEnum.DIGEST_PLAYGROUND
+      ? 'blue'
+      : 'red';
+
+  useEffect(() => {
+    setSequence(popoverData.sequence[counter.toString()] as IBeat);
+  }, [counter]);
+
+  return (
+    <Popover
+      withArrow
+      opened={sequence?.open || false}
+      transition="rotate-left"
+      transitionDuration={600}
+      position="right"
+      radius="md"
+      shadow={theme.colorScheme === 'dark' ? shadows.dark : shadows.medium}
+      offset={30}
+    >
+      <Popover.Target>
```

Comment:
let's move the popover to a reusable component, Nikolay told me that we will use it in many places later on

Reply:
Created a ticket in order to track it, NV-1831